### PR TITLE
Bundle images (jpg|png|gif) and set the public path to load files

### PR DESCRIPTION
**Pull request by pastawoua**
Local image feature for issue https://github.com/airbnb/react-sketchapp/issues/25

### CODE REVIEW

File: lib/skpm-build.js [hunk 1]
```diff
@@ -158,7 +159,10 @@ function webpackConfig (file, commandIdentifier) {
         library: commandIdentifier ? 'onRun' : undefined,
         path: commandIdentifier
           ? path.join(output, 'Contents', 'Sketch')
-          : path.join(output, 'Contents', 'Resources')
+          : path.join(output, 'Contents', 'Resources'),
+        publicPath: `file://${commandIdentifier
+          ? path.join(output, 'Contents', 'Sketch')
```
Comment (mathieudutour) [inline lib/skpm-build.js]: it's the same as above. Could you extract it in a variable and use it both in `path` and `publicPath`?

**Comment by mathieudutour** [comment]
that's really cool and seems generic enough to be included in skpm

File: lib/skpm-build.js (hunk 1 above)
Comment (pastawoua) [inline lib/skpm-build.js]: good point, updated the PR, thanks!

**Comment by mathieudutour** [comment]
actually I'm wondering if we shouldn't fork `file-loader` to change https://github.com/webpack-contrib/file-loader/blob/master/index.js#L77 to `NSImage.alloc().initWithContentsOfFile(publicPath)` so that when you require an image, you get an NSImage directly (instead of just the path) which is what you expect I guess.

We would need a change in `react-sketchapp` to handle passing directly a NSImage directly though.

What do you think @jongold?

**Comment by mathieudutour** [comment]
Also I'm not sure this approach works when distributing a plugin, I think the public path is absolute so it won't work on other machines. So we probably need to fork `file-loader` anyway to use `context.plugin.urlForResourceNamed(...).path()` or something

**Comment by jongold** [comment]
Sorry for the delayed response - awesome work @pastawoua ✨ 

it would be ~ * super rad * ~ if we didn't have to fork file-loader - I don't know enough about webpack to know how to get around that though.

can we sanity check by just trying to build a simple plugin using this branch and then running it on someone else's machine?

**Comment by pastawoua** [comment]
Yea it doesn't work, just tried on another machine, `__webpack_require__.p` is hardcoded in the build process 👎

**Comment by pastawoua** [comment]
Alright, got somewhere by removing the outputPath from the config and overwriting the `__webpack_require__.p` property with the current plugin path from the context. Not ideal though, thoughts?

```
  const scriptURL = context.scriptURL.toString();
  __webpack_require__.p = `${scriptURL.substring(0, scriptURL.lastIndexOf('/'))}/`;
```

**Comment by mathieudutour** [comment]
I think we need to use the `publicPath` option, something like this:
```
{
    ...,
   publicPath: function (url) {
        return 'context.plugin.urlForResourceNamed(' + url + ')'
   }
  ...
}
```
It does mean that context needs to be a global and there might be some tweaking to do about the `url`

**Comment by pastawoua** [comment]
@mathieudutour I've tweaked the basic-setup example https://github.com/pastawoua/react-sketchapp/tree/feature/local-image-support/examples/basic-setup

**Comment by mathieudutour** [comment]
I managed to get it to work using
```js
{
            test: /\.(jpg|png|gif)$/,
            use: {
              loader: 'file-loader',
              options: {
                outputPath: function (url) {
                  return path.join('..', 'Resources', '_webpack_images', url)
                },
                publicPath: function (url) {
                  return '"file://" + context.plugin.urlForResourceNamed("' + url.split('../Resources/')[1] + '").path()'
                }
              }
            }
          }
```

and you only need `global.context = context` in your main function

**Comment by joekim** [comment]
@mathieudutour Can you post your working example?

**Comment by mathieudutour** [comment]
Finally manage to fix this: https://github.com/skpm/skpm/commit/804046e7270ee4be45ef2db742534c9c95e1f8ca

**Comment by jongold** [comment]
Hmm, my Sketch is crashing when I try to require an image — anyone have a working example?

**Comment by mathieudutour** [comment]
I didn't publish yet, playing a bit more with it as there are a lot of changes

**Pull request closed without merging**